scm-ui/ui-webapp/src/repos/containers/RepositoryRoot.tsx

```diff
@@ -153,6 +154,38 @@ class RepositoryRoot extends React.Component<Props> {
       redirectedUrl = url + "/info";
     }
 
+    const fileControlFactoryFactory: (changeset: Changeset) => FileControlFactory = changeset => file => {
```

Comment:
This component is already way to big. You should move this function to a separate file.